lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/SpellChecker.java

```diff
@@ -51,6 +57,28 @@ public boolean spell(String word) {
     return false;
   }
 
+  private static boolean isNumber(String s) {
+    int i = 0;
+    while (i < s.length()) {
```

Comment:
Should this follow Hunspell's rules exactly? I ask because you could make it more general by scanning unicode codepoints and then verifying each codepoint's attributes (whether it is a numeric or not). I know this is a tiny fraction of use cases but unicode has all sorts of odd characters that are numerics:

https://www.fileformat.info/info/unicode/category/Nd/list.htm

Reply:
My goal is to have a drop-in replacement for Hunspell (at least for some languages), so I wouldn't generalize its behavior for now. Maybe later, and preferably accompanied by corresponding changes in native Hunspell.

Reply:
Yup, I thought so.